### ISSUE TITLE
Sluggish scrolling in notebook windows

### ISSUE THREAD

**Issue by oscarrutt** [microsoft/vscode-jupyter#4904]
Scrolling in Jupyter notebook windows is sluggish, laggy, and jumpy. It's not horrible but it is bad enough to cause stress, aggravation, and fatigue. Scrolling is nowhere as nice as in other applications when you get that "massless" feeling. JupyterLab in Firefox is an example of a Jupyter notebook that runs in a cross-platform environment that scrolls nearly perfectly. The problem appears when scrolling with either two-finger drag on the trackpad or click-dragging the thumb bar in the elevator region.

My computer runs macOS 10.11.6. This is an older version of the OS but it should have no effect on scrolling since all other apps scroll well.

I'm using Jupyter in VSCode Insiders to write Julia. I don't know if the problem extends to non-Julia notebooks.

Version: 1.53.0-insider
Commit: 8490d3dde47c57ba65ec40dd192d014fd2113496
Date: 2021-02-03T15:52:19.585Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 15.6.0

**Comment by jaymegordo** [comment]
+1 I've noticed this as well since I started using insiders + native notebooks. (both python and R)

Version: 1.54.0-insider
Commit: 11cd76005bc7516dcc726d7389d0bce1744e5c85
Date: 2021-02-22T06:44:36.656Z
Electron: 11.2.3
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin x64 20.3.0

**Comment by DavidKutu** [comment]
Thanks for letting us know @oscarrutt. I checked on a windows 10 laptop and couldn't feel a difference. I'll make sure to have someone test it with a mac.

**Comment by DavidKutu** [comment]
@oscarrutt and @jaymegordo, does this only happen with native notebooks? Does it happen say, on a regular python file in vs code?

**Comment by jaymegordo** [comment]
@DavidKutu I remember really noticing jittery/slow scrolling when I switched from "old notebooks" to native notebooks a few moths ago. In a standard large python file I can scroll top-bottom in a fraction of a second, but in a NN there's definitely a bit more resistence. I assume this is probably just related to rendering all the extra assets (tables/charts etc). I _think_ the NN scrolling has improved since I first started using it, but I cant actually compare to the old notebooks right now because they don't even work in my stable vscode (I think its something related to "experimental" settings, haven't had time or need to make it work). Its not horrible, just makes me a bit more hesitant to want to scroll around quickly.

So in summary, NN scrolling is definitely slower than a regular python file, but could just be the extra stuff to render. If I can get my old notebooks working I'll try and give a side by side comparison.

**Comment by oscarrutt** [comment]
@DavidKutu It does not happen with a regular Python file. As far as I can tell it does not happen with any text file in either vsc or vsc-i.

But to be clear, as far as I can tell, _no_ text file in either vsc or vsc-i scrolls as well as in a native Mac app. One does not have to look far: TextEdit, XCode, or even the Quick Look window in Finder. It's just that the problem as reported here is past the point of acceptability IMO.

I can (now!) open Julia native notebooks in vsc. (I thought that wasn't possible a week ago. Am I confused? The display is different, with different backgrounds and widgets but no syntax coloring in vsc.) In any case, my Julia native notebooks seem to scroll as well in vsc as text scrolls in vsc. Hope this isn't too confusing.

**Comment by DavidKutu** [comment]
Issue moved to [microsoft/vscode #117644](https://github.com/microsoft/vscode/issues/117644) via [**ZenHub**](https://www.zenhub.com/)
